From stable-bounces@linux.kernel.org  Wed Oct 11 16:04:54 2006
Date: Wed, 11 Oct 2006 18:19:24 -0400
From: Dave Jones <davej@redhat.com>
To: Greg KH <gregkh@suse.de>
Message-ID: <20061011221924.GB2248@redhat.com>
Subject: add utsrelease.h to the dontdiff file

From: Dave Jones <davej@redhat.com>

Signed-off-by: Dave Jones <davej@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/dontdiff |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.18.orig/Documentation/dontdiff
+++ linux-2.6.18/Documentation/dontdiff
@@ -135,6 +135,7 @@ tags
 times.h*
 tkparse
 trix_boot.h
+utsrelease.h*
 version.h*
 vmlinux
 vmlinux-*
