From stable-bounces@linux.kernel.org Wed Sep 27 00:48:51 2006
Date: Wed, 27 Sep 2006 03:47:49 -0400
From: Jeff Garzik <jeff@garzik.org>
To: stable@kernel.org
Message-ID: <20060927074749.GA7075@havoc.gtf.org>
Content-Disposition: inline
Subject: sata_mv: fix oops

From: Jeff Garzik <jeff@garzik.org>

From: Jeff Garzik <jeff@garzik.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/scsi/sata_mv.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.18.orig/drivers/scsi/sata_mv.c
+++ linux-2.6.18/drivers/scsi/sata_mv.c
@@ -463,6 +463,7 @@ static const struct ata_port_operations 
 
 	.qc_prep		= mv_qc_prep_iie,
 	.qc_issue		= mv_qc_issue,
+	.data_xfer		= ata_mmio_data_xfer,
 
 	.eng_timeout		= mv_eng_timeout,
 
