From stable-bounces@linux.kernel.org  Wed May 23 14:00:00 2007
Message-Id: <200705232057.l4NKvvko031767@shell0.pdx.osdl.net>
To: torvalds@linux-foundation.org
From: akpm@linux-foundation.org
Date: Wed, 23 May 2007 13:57:56 -0700
Cc: akpm@linux-foundation.org, jirislaby@gmail.com, stable@kernel.org
Subject: Char: cyclades, fix deadlock

From: Jiri Slaby <jirislaby@gmail.com>

An omitted unlock.

Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/char/cyclades.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.21.4.orig/drivers/char/cyclades.c
+++ linux-2.6.21.4/drivers/char/cyclades.c
@@ -1103,6 +1103,7 @@ static void cyy_intr_chip(struct cyclade
 
 				if (data & info->ignore_status_mask) {
 					info->icount.rx++;
+					spin_unlock(&cinfo->card_lock);
 					return;
 				}
 				if (tty_buffer_request_room(tty, 1)) {
