From stable-bounces@linux.kernel.org  Tue May 15 01:39:44 2007
Date: Tue, 15 May 2007 01:37:56 -0700 (PDT)
Message-Id: <20070515.013756.106434867.davem@davemloft.net>
To: stable@kernel.org
From: David Miller <davem@davemloft.net>
Cc: bunk@stusta.de
Subject: SPARC64: Add missing cpus_empty() check in hypervisor xcall handling.

From: "David S. Miller" <davem@sunset.davemloft.net>

Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>

---
 arch/sparc64/kernel/smp.c |    3 +++
 1 file changed, 3 insertions(+)

--- linux-2.6.21.1.orig/arch/sparc64/kernel/smp.c
+++ linux-2.6.21.1/arch/sparc64/kernel/smp.c
@@ -566,6 +566,9 @@ static void hypervisor_xcall_deliver(u64
 	unsigned long flags, status;
 	int cnt, retries, this_cpu, prev_sent, i;
 
+	if (cpus_empty(mask))
+		return;
+
 	/* We have to do this whole thing with interrupts fully disabled.
 	 * Otherwise if we send an xcall from interrupt context it will
 	 * corrupt both our mondo block and cpu list state.
