From 9e58d05a1b24d2c0471c3b4df8f473a7543d7647 Mon Sep 17 00:00:00 2001
From: Samuel Iglesias Gonsálvez <siglesias@igalia.com>
Date: Fri, 20 Jul 2012 09:39:03 +0200
Subject: Staging: ipack/bridges/tpci200: avoid kernel bug when uninstalling a device

From: Samuel Iglesias Gonsálvez <siglesias@igalia.com>

commit 9e58d05a1b24d2c0471c3b4df8f473a7543d7647 upstream.

Signed-off-by: Samuel Iglesias Gonsálvez <siglesias@igalia.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

---
 drivers/staging/ipack/bridges/tpci200.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/ipack/bridges/tpci200.c
+++ b/drivers/staging/ipack/bridges/tpci200.c
@@ -604,8 +604,8 @@ static int tpci200_slot_unregister(struc
 	if (mutex_lock_interruptible(&tpci200->mutex))
 		return -ERESTARTSYS;
 
-	ipack_device_unregister(dev);
 	tpci200->slots[dev->slot].dev = NULL;
+	ipack_device_unregister(dev);
 	mutex_unlock(&tpci200->mutex);
 
 	return 0;
