From stable-bounces@linux.kernel.org Fri Dec 30 15:26:06 2005
Date: Fri, 30 Dec 2005 01:45:12 +0100
From: Adrian Bunk <bunk@stusta.de>
To: stable@kernel.org
Message-ID: <20051230004512.GU3811@stusta.de>
Content-Disposition: inline
Cc: Jeff Garzik <jgarzik@pobox.com>, Olaf Hering <olh@suse.de>
Subject: ieee80211_crypt_tkip depends on NET_RADIO

From: Olaf Hering <olh@suse.de>

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 net/ieee80211/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.14.5.orig/net/ieee80211/Kconfig
+++ linux-2.6.14.5/net/ieee80211/Kconfig
@@ -55,7 +55,7 @@ config IEEE80211_CRYPT_CCMP
 
 config IEEE80211_CRYPT_TKIP
 	tristate "IEEE 802.11i TKIP encryption"
-	depends on IEEE80211
+	depends on IEEE80211 && NET_RADIO
 	select CRYPTO
 	select CRYPTO_MICHAEL_MIC
 	---help---
