From stable-bounces@linux.kernel.org  Wed Feb  1 21:18:58 2006
Date: Thu, 02 Feb 2006 00:09:00 -0500
From: Dmitry Torokhov <dtor_core@ameritech.net>
To: stable@kernel.org
Cc: 
Subject: Input: grip - fix crash when accessing device

Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/input/joystick/grip.c |    3 +++
 1 files changed, 3 insertions(+)

Index: linux-2.6.15.3/drivers/input/joystick/grip.c
===================================================================
--- linux-2.6.15.3.orig/drivers/input/joystick/grip.c
+++ linux-2.6.15.3/drivers/input/joystick/grip.c
@@ -192,6 +192,9 @@ static void grip_poll(struct gameport *g
 	for (i = 0; i < 2; i++) {
 
 		dev = grip->dev[i];
+		if (!dev)
+			continue;
+
 		grip->reads++;
 
 		switch (grip->mode[i]) {
