From stable-bounces@linux.kernel.org Tue Sep  5 13:15:58 2006
Date: Tue, 5 Sep 2006 22:15:08 +0200
From: Adrian Bunk <bunk@stusta.de>
To: stable@kernel.org
Message-ID: <20060905201508.GJ9173@stusta.de>
Content-Disposition: inline
Cc: Chen-Li Tien <cltien@gmail.com>, David Miller <davem@davemloft.net>
Subject: PKTGEN: Fix oops when used with balance-tlb bonding

From: Chen-Li Tien <cltien@gmail.com>

Signed-off-by: Chen-Li Tien <cltien@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Adrian Bunk <bunk@stusta.de>


---
 net/core/pktgen.c |    2 ++
 1 file changed, 2 insertions(+)

--- linux-2.6.17.11.orig/net/core/pktgen.c
+++ linux-2.6.17.11/net/core/pktgen.c
@@ -2149,6 +2149,8 @@ static struct sk_buff *fill_packet_ipv4(
 	skb->mac.raw = ((u8 *) iph) - 14 - pkt_dev->nr_labels*sizeof(u32);
 	skb->dev = odev;
 	skb->pkt_type = PACKET_HOST;
+	skb->nh.iph = iph;
+	skb->h.uh = udph;
 
 	if (pkt_dev->nfrags <= 0)
 		pgh = (struct pktgen_hdr *)skb_put(skb, datalen);
