From stable-bounces@linux.kernel.org  Mon Nov 21 02:06:57 2005
Date: Mon, 21 Nov 2005 19:07:33 +0900 (JST)
Message-Id: <20051121.190733.72492452.yoshfuji@linux-ipv6.org>
To: stable@kernel.org
From: YOSHIFUJI Hideaki <yoshfuji@linux-ipv6.org>
Cc: yoshfuji@linux-ipv6.org, netdev-core@vger.kernel.org
Subject: [PATCH] [IPV6]: Fix calculation of AH length during filling ancillary data.

From: Ville Nuorvala <vnuorval@tcs.hut.fi>

Signed-off-by: YOSHIFUJI Hideaki <yoshfuji@linux-ipv6.org>
Signed-off-by: Chris Wright <chrisw@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 net/ipv6/datagram.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.14.2.orig/net/ipv6/datagram.c
+++ linux-2.6.14.2/net/ipv6/datagram.c
@@ -437,7 +437,7 @@ int datagram_recv_ctl(struct sock *sk, s
 				break;
 			case IPPROTO_AH:
 				nexthdr = ptr[0];
-				len = (ptr[1] + 1) << 2;
+				len = (ptr[1] + 2) << 2;
 				break;
 			default:
 				nexthdr = ptr[0];
