From stable-bounces@linux.kernel.org  Wed Jun 29 20:42:38 2005
Date: Thu, 30 Jun 2005 11:42:10 +0800
From: Wen-chien Jesse Sung <jesse@cola.voip.idv.tw>
To: stable@kernel.org
Subject: [PATCH] fix semaphore handling in __unregister_chrdev_region

This up() should be down() instead.

Signed-off-by: Wen-chien Jesse Sung <jesse@cola.voip.idv.tw>
Signed-off-by: Chris Wright <chrisw@osdl.org>
---

Index: linux-2.6.12.y/fs/char_dev.c
===================================================================
--- linux-2.6.12.y.orig/fs/char_dev.c
+++ linux-2.6.12.y/fs/char_dev.c
@@ -139,7 +139,7 @@ __unregister_chrdev_region(unsigned majo
 	struct char_device_struct *cd = NULL, **cp;
 	int i = major_to_index(major);
 
-	up(&chrdevs_lock);
+	down(&chrdevs_lock);
 	for (cp = &chrdevs[i]; *cp; cp = &(*cp)->next)
 		if ((*cp)->major == major &&
 		    (*cp)->baseminor == baseminor &&
