From stable-bounces@linux.kernel.org Tue Jul 31 00:47:11 2007
From: Alexey Dobriyan <adobriyan@sw.ru>
Date: Tue, 31 Jul 2007 00:38:50 -0700
Subject: Fix leak on /proc/lockdep_stats
To: torvalds@linux-foundation.org
Cc: adobriyan@sw.ru, akpm@linux-foundation.org, stable@kernel.org
Message-ID: <200707310738.l6V7co6e022182@imap1.linux-foundation.org>


From: Alexey Dobriyan <adobriyan@sw.ru>

Signed-off-by: Alexey Dobriyan <adobriyan@sw.ru>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 kernel/lockdep_proc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/lockdep_proc.c
+++ b/kernel/lockdep_proc.c
@@ -339,7 +339,7 @@ static const struct file_operations proc
 	.open		= lockdep_stats_open,
 	.read		= seq_read,
 	.llseek		= seq_lseek,
-	.release	= seq_release,
+	.release	= single_release,
 };
 
 static int __init lockdep_proc_init(void)
