From stable-bounces@linux.kernel.org Fri Aug 15 12:05:14 2008
From: Steve French <sfrench@us.ibm.com>
Date: Fri, 15 Aug 2008 19:05:06 GMT
Subject: CIFS: if get root inode fails during mount, cleanup tree connection
To: jejb@kernel.org, stable@kernel.org
Message-ID: <200808151905.m7FJ56sc001284@hera.kernel.org>

From: Steve French <sfrench@us.ibm.com>

commit 2c731afb0d4ba16018b400c75665fbdb8feb2175 upstream

Signed-off-by: Steve French <sfrench@us.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/cifs/cifsfs.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/cifs/cifsfs.c
+++ b/fs/cifs/cifsfs.c
@@ -175,6 +175,8 @@ out_no_root:
 	if (inode)
 		iput(inode);
 
+	cifs_umount(sb, cifs_sb);
+
 out_mount_failed:
 	if (cifs_sb) {
 #ifdef CONFIG_CIFS_DFS_UPCALL
