From b7d40315c9643034ac4b5c9dda480d0124416f89 Mon Sep 17 00:00:00 2001
From: Michael Chan <mchan@broadcom.com>
Date: Wed, 13 Jul 2011 17:24:18 +0000
Subject: cnic: Fix port_mode setting

From: Michael Chan <mchan@broadcom.com>

commit b7d40315c9643034ac4b5c9dda480d0124416f89 upstream.

CHIP_2_PORT_MODE was not set correctly.

Signed-off-by: Michael Chan <mchan@broadcom.com>
Reviewed-by: Matt Carlson <mcarlson@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/net/cnic.c
+++ b/drivers/net/cnic.c
@@ -4834,7 +4834,7 @@ static int cnic_start_bnx2x_hw(struct cnic_dev *dev)
 			cp->port_mode = CHIP_4_PORT_MODE;
 			cp->pfid = func >> 1;
 		} else {
-			cp->port_mode = CHIP_4_PORT_MODE;
+			cp->port_mode = CHIP_2_PORT_MODE;
 			cp->pfid = func & 0x6;
 		}
 	} else {
